bin/assignUsersToPersonalizedDigestVariation.ts

```diff
@@ -0,0 +1,66 @@
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import fs from 'fs';
+import { parse } from 'csv-parse';
+
+(async (): Promise<void> => {
+  const csvFilePath = process.argv[2];
+  const variationArgument = process.argv[3];
+  const splitBy = process.argv[4] || ',';
+
+  if (!csvFilePath) {
+    throw new Error('CSV file path is required');
+  }
+
+  const variation = +variationArgument;
+
+  if (Number.isNaN(variation) || variation < 1) {
+    throw new Error(
+      'variation argument is invalid, it should be positive number',
+    );
+  }
+
+  const userIds: string[] = [];
+
+  console.log('reading csv file', csvFilePath);
+
+  const stream = fs
+    .createReadStream(csvFilePath)
+    .pipe(parse({ delimiter: splitBy, from_line: 2 }));
+
+  stream.on('error', (err) => {
+    console.error('failed to read file: ', err.message);
+  });
+
+  stream.on('data', function ([userId]) {
+    if (!userId) {
+      return;
+    }
+
+    userIds.push(userId);
+  });
+
+  await new Promise((resolve) => {
+    stream.on('end', resolve);
+  });
+
+  console.log('running db query');
+
+  const con = await createOrGetConnection();
+
+  await con.transaction(async (manager) => {
+    await manager.query(
+      `
+        UPDATE user_personalized_digest
```

Comment:
This will be one big query. I suggest splitting it into batches of 1k users roughly

Reply:
Had the batch logic, I thought I'll just chunk the CSV?

Reply:
Sure that works!

Reply:
Started to split up CSV.. figured its easier to just batch so.. added batching 0f607d889a4f9e8bf51bbd7a0ff7dac21eb31aaf

Reply:
I tried 10k with initial import and it went instantly so I think it is ok default.